### ISSUE TITLE
clippy::large_enum_variant should suggest Option<Box<T>> over Box<Option<T>>

### ISSUE THREAD

**Issue by frxstrem** [rust-lang/rust-clippy#9983]
### Description

Consider this example:

```rust
struct LargeValue([u8; 201]);

enum Foo {
    Large(Option<LargeValue>),
    Small,
}
```

Running `cargo clippy` gives this warning:

```
warning: large size difference between variants
 --> src/lib.rs:3:1
  |
3 | / enum Foo {
4 | |     Large(Option<LargeValue>),
  | |     ------------------------- the largest variant contains at least 202 bytes
5 | |     Small,
  | |     ----- the second-largest variant carries no data at all
6 | | }
  | |_^ the entire enum is at least 202 bytes
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant
  = note: `#[warn(clippy::large_enum_variant)]` on by default
help: consider boxing the large fields to reduce the total size of the enum
  |
4 |     Large(Box<Option<LargeValue>>),
  |           ~~~~~~~~~~~~~~~~~~~~~~~
```

Boxing an option is inefficient in the case of `None`, since it still needs to allocate the `None` on the heap and it can't take advantage of [null pointer optimization](https://doc.rust-lang.org/std/option/#representation). Instead, the lint should suggest <code>Option&lt;Box&lt;<i>LargeValue</i>&gt;&gt;</code> when the original field is <code>Option&lt;<i>LargeValue</i>&gt;</code>.

### Version

```text
rustc 1.67.0-nightly (a28f3c88e 2022-11-20)
binary: rustc
commit-hash: a28f3c88e50a77bc2a91889241248c4543854e61
commit-date: 2022-11-20
host: x86_64-apple-darwin
release: 1.67.0-nightly
LLVM version: 15.0.4
```

**Comment by Centri3** [comment]
Opened #10960, which while it doesn't cover changing the suggestion should work
